.github/workflows/release.yaml

```diff
@@ -0,0 +1,40 @@
+# Release workflow: Pushing a tag triggers this workflow,
+# which uses the name of the tag as the container tag/version
+# that goes into GHCR.
+name: Release
+
+# Build and publish Docker image upon pushing a tag
+on:
+  push:
+    tags:
+      - "v*"
+
+env:
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        # actions/checkout@v2 breaks annotated tags by converting them into
+        # lightweight tags, so we need to force fetch the tag again
+        # See: https://github.com/actions/checkout/issues/290
+      - name: "Repair tag"
+        run: git fetch -f origin ${{ github.ref }}:${{ github.ref }}
+      - name: "Verify tag is annotated"
+        run: if test x$(git for-each-ref ${{ github.ref }} | awk '{print $2}') = xtag; then /bin/true; else echo "\"${{ github.ref }}\" does not look like an annotated tag!"; /bin/false; fi
+      - name: "Echo release tag"
+        run: echo "TAG=${{ github.repository }}:${{ github.ref }}"
+      - name: "Log in to Docker"
+        uses: docker/login-action@v1
+        with:
+          password: ${{ secrets.GITHUB_TOKEN }}
+          registry: ghcr.io
+          username: ${{ github.actor }}
+      - id: docker_build
+        name: "Build the release Docker image"
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          tags: ghcr.io/${IMAGE_NAME}:${{ github.ref }}
```

Comment:
Negative:



`github.ref` | string | The fully-formed ref of the branch or tag that triggered the workflow run. For workflows triggered by push, this is the branch or tag ref that was pushed. For workflows triggered by pull_request, this is the pull request merge branch. For workflows triggered by release, this is the release tag created. For other triggers, this is the branch or tag ref that triggered the workflow run. This is only set if a branch or tag is available for the event type. The ref given is fully-formed, meaning that for branches the format is refs/heads/<branch_name>, for pull requests it is refs/pull/<pr_number>/merge, and for tags it is refs/tags/<tag_name>. For example, refs/heads/feature-branch-1.
-- | -- | --

Because this workflow is only triggered by a _push_ to a tag starting with "v*", the ref would be the tag that triggered the workflow run.